core/src/css/test/theme-ionic/typography.e2e.ts

```diff
@@ -0,0 +1,185 @@
+          <p><ion-text class="ionic-heading4">Utility class 'ionic-heading4'</ion-text></p>
+          <p><ion-text class="ionic-heading5">Utility class 'ionic-heading5'</ion-text></p>
+          <p><ion-text class="ionic-heading6">Utility class 'ionic-heading6'</ion-text></p>
+        </ion-content>
+      `,
+        config
+      );
+
+      const globalStyles = page.locator('ion-content');
+      await expect(globalStyles.nth(0)).toHaveScreenshot(screenshot(`ionic-utility-classes-headings`));
+    });
+
+    test('should apply the utility classes for font weights', async ({ page }) => {
+      await page.setContent(
+        `
+        <link href="../../../../../css/ionic.globals.ionic.css" rel="stylesheet" />
+        <style>
+          .ionic-preview {
+            background: #d7d8da;
+            border-radius: 16px;
+            box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
+            font-size: 16px;
+            font-weight: 100;
+            margin-bottom: 8px;
+            padding: 8px;
+          }
+        </style>
+
+        <ion-content class="ion-padding" id="content">
+          <ion-text class="ionic-preview">Font Weights</ion-text>
+          <p><ion-text class="ionic-font-light">Utility class 'ionic-font-light'</ion-text></p>
+          <p><ion-text class="ionic-font-regular">Utility class 'ionic-font-regular'</ion-text></p>
+          <p><ion-text class="ionic-font-medium">Utility class 'ionic-font-medium'</ion-text></p>
+          <p><ion-text class="ionic-font-semibold">Utility class 'ionic-font-semibold'</ion-text></p>
+          <p><ion-text class="ionic-font-bold">Utility class 'ionic-font-bold'</ion-text></p>
+        </ion-content>
+      `,
+        config
+      );
+
+      const globalStyles = page.locator('ion-content');
+      await expect(globalStyles.nth(0)).toHaveScreenshot(screenshot(`ionic-utility-classes-font-weights`));
+    });
+
+    test('should apply the utility classes for font display', async ({ page }) => {
+      await page.setContent(
+        `
+        <link href="../../../../../css/ionic.globals.ionic.css" rel="stylesheet" />
+        <style>
+          .ionic-preview {
+            background: #d7d8da;
+            border-radius: 16px;
+            box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
+            font-size: 16px;
+            font-weight: 100;
+            margin-bottom: 8px;
+            padding: 8px;
+          }
+        </style>
+
+        <ion-content class="ion-padding" id="content">
+          <ion-text class="ionic-preview">Font Displays</ion-text>
+          <p><ion-text class="ionic-display-s">Utility class 'ionic-display-s'</ion-text></p>
+          <p><ion-text class="ionic-display-m">Utility class 'ionic-display-m'</ion-text></p>
+        </ion-content>
+      `,
+        config
+      );
+
+      const globalStyles = page.locator('ion-content');
+      await expect(globalStyles.nth(0)).toHaveScreenshot(screenshot(`ionic-utility-classes-font-displays`));
```

Comment:
_You may need to branch off of this repository and merge your fork into that branch before running these steps._

Follow [these steps](https://github.com/ionic-team/ionic-framework/blob/main/core/src/utils/test/playwright/docs/usage-instructions.md#generating-or-updating-ground-truths-ci) to generate the screenshots for this test on CI: 

- Navigate to the [Update Reference Screenshots GitHub Action](https://github.com/ionic-team/ionic-framework/actions/workflows/update-screenshots.yml)
- Click the **Run workflow** dropdown.
- Select your branch.
- Enter a path or component name(s) to run the action for specific tests instead of all. For example, you can pass this path: `src/css/test/theme-ionic/typography.e2e.ts` to generate screenshots for only this test.
- Click **Run workflow**.

This will generate the screenshots for this test and commit them to your branch. Make sure you have updated the config to only run for the `ionic` theme first or it will generate them for `ios` and `md` too. 

This can also be done locally by running the Docker commands that Liam will show tomorrow, if you would like to wait and add them that way. Let me know if you have any questions!

Reply:
I highly recommend updating the screenshots after changing the folder name as [requested](https://github.com/ionic-team/ionic-framework/pull/29212#discussion_r1548518005). Otherwise, you'll end up having to update them again.